lib/pet_store_web/controllers/user_settings_controller.ex

```diff
@@ -8,6 +8,31 @@ defmodule PetStoreWeb.UserSettingsController do
 
   action_fallback PetStoreWeb.FallbackController
 
+  # Updates email address for lower-admin_level accounts
+  def update(conn, %{"action" => "update_email", "target" => target_email, "value" => new_email}) do
+    with {:ok, target} <-
+           Accounts.maybe_redacted_user_by_email(target_email, conn.assigns.current_user),
+         email_changeset = Accounts.change_user_email(target, %{email: new_email}),
+         {:ok, new_user} <- Ecto.Changeset.apply_action(email_changeset, :update) do
```

Comment:
You shouldn't call `Ecto` from the Controller layer. I see there's an `apply_user_email` below, either use that or provide some other similar helper from the context.

Reply:
Now that I have a look at this what you probably want to do is adding an additional authz call (e.g. using Bodyguard) before `apply_user_email`. That way you don't even need `maybe_redacted_user_by_email`.

Reply:
well yeah the whole point of `maybe_redacted_user_by_email` is to provide authentication to `fetch_user_by_email`. For now I'll add a new helper function (I still don't see much of a gain in using Bodyguard), then if you insist that's the better approach I'll proceed with a refactor

Reply:
Done in https://github.com/noaccOS/Pet-Store/pull/4/commits/e0615b05e72577553d599dbaccd9bece3e5f031d